src/workerd/jsg/modules.h

```diff
@@ -252,12 +260,29 @@ class ModuleRegistryImpl final: public ModuleRegistry {
   // object by identity. We use a kj::Table!
   struct Entry {
     kj::Path specifier;
-    ModuleInfo info;
+    kj::OneOf<ModuleInfo, kj::ArrayPtr<const char>> info;
 
-    Entry(kj::Path& specifier, ModuleInfo info)
+    Entry(const kj::Path& specifier, ModuleInfo info)
         : specifier(specifier.clone()), info(kj::mv(info)) {}
+
+    Entry(const kj::Path& specifier, kj::ArrayPtr<const char> src)
+        : specifier(specifier.clone()), info(src) {}
+
     Entry(Entry&&) = default;
     Entry& operator=(Entry&&) = default;
+
+    ModuleInfo& module(v8::Isolate* isolate) {
+      KJ_SWITCH_ONEOF(info) {
+        KJ_CASE_ONEOF(moduleInfo, ModuleInfo) {
+          return moduleInfo;
+        }
+        KJ_CASE_ONEOF(src, kj::ArrayPtr<const char>) {
+          info = ModuleInfo(isolate, specifier.basename()[0], src);
```

Comment:
(to be clear, just saying that a comment here would be good :-) ...)

Reply:
To import node core modules it should be `require('node:... ')` where `... ` is the name of the core module. E.g. `require('node:assert')`

Reply:
I don't think this name is relate to how you import. This name goes to ScriptOrigin.

I don't know if it should be the same or not.

PS nit: only ESM import will be supported, not require.

Reply:
Why wouldn't require be supported? This should work for that also

Reply:
> Why wouldn't require be supported?

Because then we'll have to compile to version of the modules? One for ESM and one for commonjs?

Or can ESM modules be imported safely using require too?

Reply:
The way we handle imports and require it should just work. Worth testing though

Reply:
(to be clear, it *might* not just work also)